internal/flag/rule.go

```diff
@@ -240,8 +240,11 @@ func (r *Rule) IsValid(defaultRule bool) error {
 	// Validate the percentage of the rule
 	if r.Percentages != nil {
 		count := float64(0)
-		for _, p := range r.GetPercentages() {
+		for k, p := range r.GetPercentages() {
 			count += p
+			if _, ok := variations[k]; !ok {
+				return fmt.Errorf("invalid percentage: variation %s does not exists", k)
```

Comment:
It might be better to change like this?
```suggestion
				return fmt.Errorf("invalid percentage: variation %s does not exist", k)
```